rpc/block_test.go

```diff
@@ -353,6 +351,279 @@ func TestBlockWithTxsAndInvokeTXNV0(t *testing.T) {
+				LookupTxnPositionInExpected: 0,
+				LookupTxnPositionInOriginal: 0,
+				ExpectedError:               nil,
+				want:                        &fullBlockSepolia30436,
+			},
+			{
+				BlockID:                     WithBlockNumber(30436),
+				LookupTxnPositionInExpected: 0,
+				LookupTxnPositionInOriginal: 0,
+				ExpectedError:               nil,
+				want:                        &fullBlockSepolia30436,
+			},
+		},
+		"mainnet": {},
+	}[testEnv]
+
+	for _, test := range testSet {
+		spy := NewSpy(testConfig.provider.c)
+		testConfig.provider.c = spy
+		blockWithTxsInterface, err := testConfig.provider.BlockWithTxs(context.Background(), test.BlockID)
+		if err != test.ExpectedError {
+			t.Fatal("BlockWithTxHashes match the expected error:", err)
+		}
+		if test.ExpectedError != nil && blockWithTxsInterface == nil {
+			continue
+		}
+		blockWithTxs, ok := blockWithTxsInterface.(*Block)
+		if !ok {
+			t.Fatalf("expecting *rpv02.Block, instead %T", blockWithTxsInterface)
+		}
+		_, err = spy.Compare(blockWithTxs, false)
+		if err != nil {
+			t.Fatal("expecting to match", err)
+		}
+		if !strings.HasPrefix(blockWithTxs.BlockHash.String(), "0x") {
+			t.Fatal("Block Hash should start with \"0x\", instead", blockWithTxs.BlockHash)
+		}
+
+		if len(blockWithTxs.Transactions) == 0 {
+			t.Fatal("the number of transaction should not be 0")
+		}
+
+		if test.want != nil {
+			if (*test.want).BlockHash == &felt.Zero {
+				continue
+			}
+
+			invokeV1Want, ok := (*test.want).Transactions[test.LookupTxnPositionInExpected].(BlockInvokeTxnV1)
+			if !ok {
+				t.Fatal("expected invoke v1 transaction")
+			}
+			invokeV1Block, ok := blockWithTxs.Transactions[test.LookupTxnPositionInOriginal].(BlockInvokeTxnV1)
+			if !ok {
+				t.Fatal("expected invoke v1 transaction")
+			}
+			require.Equal(t, invokeV1Want.TransactionHash, invokeV1Block.TransactionHash, "expected equal TransactionHash")
+			require.Equal(t, invokeV1Want.InvokeTxnV1.MaxFee, invokeV1Block.InvokeTxnV1.MaxFee, "expected equal maxfee")
+			require.Equal(t, invokeV1Want.InvokeTxnV1.SenderAddress, invokeV1Block.InvokeTxnV1.SenderAddress, "expected equal senders addresses")
+		}
+	}
+}
+
+// TestBlockWithTxsAndInvokeTXNV3 tests the BlockWithTxsAndInvokeTXNV3 function.
+//
+// The function tests the BlockWithTxsAndInvokeTXNV3 function by setting up a test configuration and a test set type.
+// It then initializes a fullBlockSepolia52767 variable with a Block struct and invokes the BlockWithTxs function with different test scenarios.
+// The function compares the expected error with the actual error and checks if the BlockWithTxs function returns the correct block data.
+// It also verifies the block hash, the number of transactions in the block, and the details of a specific transaction.
+//
+// Parameters:
+// - t: The t testing object
+// Returns:
+//
+//	none
+func TestBlockWithTxsAndInvokeTXNV3(t *testing.T) {
```

Comment:
The `TestBlockWithTxsAndInvokeTXNV3` must follow the existing [TestBlockWithTxsAndInvokeTXNV1 ](https://github.com/NethermindEth/starknet.go/blob/main/rpc/block_test.go#L703)test because it was already updated with the requirements specified in these comments https://github.com/NethermindEth/starknet.go/pull/566#discussion_r1598451390

Reply:
I've pushed the updated test, which is now passing locally :D